eng/pipelines/templates/stages/archetype-sdk-client.yml

```diff
@@ -0,0 +1,281 @@
+        # previous step. This is only generated in non-PR builds (i.e. nightly runs, etc)
+        - task: Maven@3
+          displayName: 'Run SpotBugs'
+          condition: ne(variables['Build.Reason'], 'PullRequest')
+          inputs:
+            mavenPomFile: eng/spotbugs-aggregate-report/pom.xml
+            options: '--batch-mode -Djava-lts -DskipTests -Dgpg.skip'
+            mavenOptions: '$(LoggingOptions)'
+            javaHomeOption: 'JDKVersion'
+            jdkVersionOption: '1.11'
+            jdkArchitectureOption: 'x64'
+            publishJUnitResults: false
+            goals: 'install'
+
+        - script: |
+            git clone https://github.com/JonathanGiles/DependencyChecker.git
+            mkdir input && cp dependencies.json input/
+          displayName: 'Download dependency checker'
+
+        - task: Maven@3
+          displayName: 'Install azure-sdk-parent'
+          inputs:
+            mavenPomFile: parent/pom.xml
+            options: '$(DefaultOptions) -DskipTests -Dgpg.skip'
+            mavenOptions: '$(LoggingOptions)'
+            javaHomeOption: 'JDKVersion'
+            jdkVersionOption: '1.11'
+            jdkArchitectureOption: 'x64'
+            publishJUnitResults: false
+            goals: 'install'
+
+        - task: Maven@3
+          displayName: 'Analyze dependencies'
+          inputs:
+            mavenPomFile: 'DependencyChecker/pom.xml'
+            options: '-Dexec.args="-showall -dependencymanagement"'
+            mavenOptions: '$(LoggingOptions)'
+            javaHomeOption: 'JDKVersion'
+            jdkVersionOption: '1.11'
+            jdkArchitectureOption: 'x64'
+            publishJUnitResults: false
+            goals: 'clean package exec:java'
+
+        # We only copy artifacts into the staging in nightly builds, we don't bother with this for PR validation builds.
+        - powershell: |
+            copy -r target/staging $(Build.ArtifactStagingDirectory)
+            copy eng/code-quality-reports/src/main/resources/index.html $(Build.ArtifactStagingDirectory)
+            copy eng/spotbugs-aggregate-report/target/spotbugs/spotbugsXml.html (Join-Path $(Build.ArtifactStagingDirectory) "staging")
+            copy output/dependencies.html (Join-Path $(Build.ArtifactStagingDirectory) "staging")
+          displayName: 'Copy reports to artifact staging'
+          condition: ne(variables['Build.Reason'], 'PullRequest')
+
+        - task: PublishBuildArtifacts@1
+          condition: succeededOrFailed()
+          displayName: 'Publish Report Artifacts'
+          inputs:
+            artifactName: reports
+
+
+    - job: 'Test'
+
+      variables:
+        - template: ../variables/globals.yml
+
+      strategy:
+        matrix:
+          Linux - Java 8:
+            OSName: 'Linux'
+            OSVmImage: 'ubuntu-16.04'
+            JavaVersion: '1.8'
+            ProfileFlag: ''
+          macOS - Java 8:
+            OSName: 'macOS'
+            OSVmImage: 'macOS-10.13'
```

Comment:
is upgrading to 10.14 images on our radar soon?

Reply:
We didn't have anything planned yet, but we can if there is a need?

Reply:
We do have a work item to update Windows build agents: https://github.com/azure/azure-sdk/issues/527

I've opened a work item to upgrade to 10.14: https://github.com/azure/azure-sdk/issues/527